tv_apps/fling-player/js/fling_player.js

```diff
@@ -164,79 +188,181 @@
         sec < 0 ||
         sec > duration
     ) {
-      mDBG.warn('Not moving due to corrupt type or sec!');
+      mDBG.warn('Not moving due to corrupt type/sec', type, sec);
       return;
     }
 
-    mDBG.log('Move to ', (100 * sec / duration) + '%');
+    requestAnimationFrame(() => {
+      mDBG.log('Move to sec / duration = %d / %d', sec, duration);
+      timeBar.style.width = (100 * sec / duration) + '%';
+    });
+  };
+
+  /**
+   * @param {string} type 'elapsed' or 'duration'
+   * @param {number} sec
+   */
+  proto.writeTimeInfo = function (type, sec) {
+
+    var timeInfo = this[`_${type}Time`];
+    var duration = this._player.getRoundedDuration();
+    sec = Math.round(sec);
+
+    if (!timeInfo ||
+        (sec >= 0) === false ||
+        (sec <= duration) === false
+    ) {
+      return;
+    }
 
-    timeBar.style.width = (100 * sec / duration) + '%';
+    var t = this._player.parseTime(sec);
+    timeInfo.textContent = (t.hh <= 0) ?
+                            t.mm + ':' + t.ss :
+                            t.hh + ':' + t.mm + ':' + t.ss;
   };
 
   // UI handling end
 
   // Video handling
 
-  proto._startAutoSeek = function fp_startAutoSeek(dir) {
+  proto.play = function () {
+    this._player.play();
+    this.setPlayButtonState('playing');
+  };
 
-    if (this._autoSeekStartTime == null) { // Do not double start
+  proto.pause = function () {
+    this._player.pause();
+    this.setPlayButtonState('paused');
+  };
 
-      this._autoSeekStartTime = (new Date()).getTime();
-      this._autoSeekDirection = dir;
-      this._autoSeek();
+  proto._startUpdateControlPanelContinuously = function () {
+
+    if (this._contrlPanelUpdateTimer == null) {
+
+      // To stop this is important.
+      // We don't want they get messed with each other
+      this._stopSeekOnKeyPress();
+
+      this._contrlPanelUpdateTimer = setTimeout(
+        this._updateControlPanelContinuously.bind(this)
+      );
+      this.showControlPanel(true);
     }
   };
 
-  proto._stopAutoSeek = function fp_stopAutoSeek(dir) {
-    this._autoSeekStartTime = null;
-    this._autoSeekDirection = null;
+  proto._stopUpdateControlPanelContinuously = function () {
```

Comment:
nit: I still feel the function name is not good enough, but let's not worry about it now and move forward